src/trg-client.c

```diff
@@ -615,6 +758,15 @@ static CURL* get_curl(TrgClient *tc, guint http_class)
                 curl_easy_setopt(curl, CURLOPT_PASSWORD, password);
             }
             curl_easy_setopt(curl, CURLOPT_URL, trg_client_get_url(tc));
+
+#ifdef HAVE_MICRODNS
+            if (priv->local_hosts) {
+                    struct curl_slist *hosts = NULL;
+
+                    hosts = curl_slist_append(hosts, priv->local_hosts);
+                    curl_easy_setopt(curl, CURLOPT_RESOLVE, hosts);
```

Comment:
This appears to leak `hosts`. According to the docs it must be freed: https://curl.se/libcurl/c/CURLOPT_RESOLVE.html